cryptoki/src/context/general_purpose.rs

```diff
@@ -120,6 +121,81 @@ pub enum Function {
     WaitForSlotEvent,
 }
 
+impl Display for Function {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
```

Comment:
:thinking: quite a lot of code... is there a reason you didn't just use `write!(f, "Function::{:?}", self)`? (Maybe you didn't want to depend on the Debug trait's derive that just outputs variant's name?)